src/main.c

```diff
@@ -2977,6 +2978,11 @@ int main(int argc, char **argv)
     cmd_argc = argc;
     cmd_argv = argv;
 
+#if !defined(LIBRESSL_VERSION_NUMBER)
+    /* assert that openssl library and header versions are the same */
+    assert(OpenSSL_version_num() == OPENSSL_VERSION_NUMBER);
```

Comment:
```suggestion
    assert(SSLeay() == OPENSSL_VERSION_NUMBER);
```

Looks like OpenSSL 1.0.2g does not provide `OpenSSL_version_num()`. It was called `SSLeay()` and is still available as a macro.

Reply:
thanks @gfx. Wow, so I guess h2o supports openssl 1.0.0x?